src/components/KpiReact/index.js

```diff
@@ -0,0 +1,186 @@
+      );
+      recovered.percentage = t("recovered-percentage", {
+        percent: formatNumber(recoveredPercent),
+      });
+    }
+    setDrawRecovered(recovered);
+  }, [data, language, t]);
+  useEffect(() => {
+    const { totals, totalsDiff } = data;
+    const formatNumber = maybeIntlNumberFormat(language);
+
+    let deaths = {
+      id: "deceased",
+      label: t("kpi-deceased"),
+      value: formatNumber(totals.deceased),
+      diff:
+        totalsDiff.deceased >= 0
+          ? `+${formatNumber(totalsDiff.deceased)}`
+          : formatNumber(totalsDiff.deceased),
+      chartName: "deceased_daily_avg.svg",
+      caption: t("deceased-chart-caption"),
+    };
+    if (totals.confirmed > 0) {
+      let deceasedPercent = parseInt(
+        (totals.deceased / totals.confirmed) * 100
+      );
+      if (deceasedPercent < 1) {
+        deceasedPercent =
+          parseInt((totals.deceased / totals.confirmed) * 1000) / 10;
+      }
+      deaths.percentage = t("deceased-percentage", {
+        percent: formatNumber(deceasedPercent),
+      });
+    }
+    setDrawDeaths(deaths);
+  }, [data, language, t]);
+  useEffect(() => {
+    const { totals, totalsDiff } = data;
+    const formatNumber = maybeIntlNumberFormat(language);
+
+    let active = {
+      id: "active",
+      label: t("kpi-active"),
+      value: formatNumber(totals.active),
+      diff:
+        totalsDiff.active >= 0
+          ? `+${formatNumber(totalsDiff.active)}`
+          : formatNumber(totalsDiff.active),
+      chartName: "active_cumulative_avg.svg",
+      caption: t("active-chart-caption"),
+      isActive: true,
+    };
+    if (totals.active > 0) {
+      let criticalPercentage = parseInt(
+        (totals.critical / totals.active) * 100
+      );
+      if (criticalPercentage < 1) {
+        criticalPercentage =
+          parseInt((totals.critical / totals.active) * 1000) / 10;
+      }
+      active.percentage = t("active-critical-percentage", {
+        percent: formatNumber(criticalPercentage),
+      });
+    }
+
+    setDrawActive(active);
+  }, [data, language, t]);
+  useEffect(() => {
+    const { totals, totalsDiff } = data;
+    const formatNumber = maybeIntlNumberFormat(language);
+
+    let tested = {
+      id: "tested",
+      label: t("kpi-tested"),
+      value: formatNumber(totals.tested),
+      diff:
+        totalsDiff.tested >= 0
+          ? `+${formatNumber(totalsDiff.tested)}`
+          : formatNumber(totalsDiff.tested),
+      chartName: "tested_daily_avg.svg",
+      caption: t("tested-chart-caption"),
+    };
+
+    if (totals.tested > 0) {
+      let testedPercentage = parseInt((totals.confirmed / totals.tested) * 100);
+      if (testedPercentage < 1) {
+        testedPercentage =
+          parseInt((totals.confirmed / totals.tested) * 1000) / 10;
+      }
+      tested.percentage = t("tested-percentage", {
+        percent: formatNumber(testedPercentage),
+      });
+    }
+    setDrawTested(tested);
```

Comment:
Why do you need to put this logic inside of useEffect and store in state chart data for all types if it renders in a loop for a single type, plus if data comes outside of React and data is already loaded on the moment it mount in DOM and there is no way it will change inside of component? Aggregating of chart data for each kpi-type looks the same, it cold be moved to method to reduce repeated code for example something like this 
```
const chart = {
      id: type,
      label: t(`kpi-${type}`),
      value: formatNumber(totals[type]),
      diff:
        totalsDiff[type] >= 0
          ? `+${formatNumber(totalsDiff[type])}`
          : formatNumber(totalsDiff[type]),
      ...
```